app/uk/gov/hmrc/nisp/views/includes/nirecordtaxyear.scala.html

```diff
@@ -80,16 +80,18 @@
                     @if(niRecordTaxYear.classOneContributions <= 0 && niRecordTaxYear.otherCredits <= 0 && niRecordTaxYear.classThreeCredits <= 0 && niRecordTaxYear.classTwoCredits <= 0){
                         <p class="contributions-header">@Messages("nisp.nirecord.youdidnotmakeanycontrib")</p>
                     }
-                    <p>@Html(Messages("nisp.nirecord.gap.findoutmoreabout",routes.NIRecordController.showGapsAndHowToCheckThem().url))</p>
+                        <p>@Html(Messages("nisp.nirecord.gap.findoutmoreabout",routes.NIRecordController.showGapsAndHowToCheckThem().url))</p>
 
                     @if(niRecordTaxYear.payable && niRecordTaxYear.classThreePayableBy.isDefined) {
-                      @niRecordTaxYear.classThreePayableBy.map { classThreePayableByDate =>
-                        <p class="contributions-header">@Messages("nisp.nirecord.gap.youcanmakeupshortfall")</p>
-                        <p>@Html(Messages("nisp.nirecord.gap.payvoluntarycontrib",NispMoney.pounds(niRecordTaxYear.classThreePayable.getOrElse(0)),Dates.formatDate(classThreePayableByDate.localDate)))</p>
-                        <p>@Html(Messages("nisp.nirecord.gap.findoutmore",routes.NIRecordController.showVoluntaryContributions().url))</p>
-                      }
-                    }else{
-                    <p class="panel-indent">@Messages("nisp.nirecord.gap.latepaymentmessage")</p>
+                        @niRecordTaxYear.classThreePayableBy.map { classThreePayableByDate =>
+                        @niRecordTaxYear.classThreePayableByPenalty.map { classThreePayableWithPenalty =>
+                            <p class="contributions-header">@Messages("nisp.nirecord.gap.youcanmakeupshortfall")</p>
+                            <p>@Html(Messages("nisp.nirecord.gap.payvoluntarycontrib",NispMoney.pounds(niRecordTaxYear.classThreePayable.getOrElse(0)),Dates.formatDate(classThreePayableWithPenalty.localDate),Dates.formatDate(classThreePayableByDate.localDate)))</p>
+                            <p>@Html(Messages("nisp.nirecord.gap.findoutmore",routes.NIRecordController.showVoluntaryContributions().url))</p>
+                        }
+                        }
```

Comment:
There shouldn't be such a scenario, but if there is the maps (or the for comprehension) should hide the text if one is missing. Maybe worth checking, as it might change in future years?